File: solutions/elixir/grains/1/lib/grains.ex
defmodule Grains do
  @doc """
  Calculate two to the power of the input minus one.
  """
  @spec square(pos_integer()) :: {:ok, pos_integer()} | {:error, String.t()}
  def square(number) do
    exponent = number - 1

    if exponent >= 0 and exponent < 64 do
      {:ok, 2 ** exponent}
    else
      {:error, "The requested square must be between 1 and 64 (inclusive)"}
    end
  end

  @doc """
  Adds square of each number from 1 to 64.
  """
  @spec total :: {:ok, pos_integer()}
  def total do
    sum =
      Enum.map(1..64, fn n ->
        {:ok, square} = square(n)
        square
      end)
      |> Enum.sum()

    {:ok, sum}
  end
end

File: solutions/typescript/grains/1/grains.ts
export const square = (sq: number): bigint => {
  if ( sq <= 0 || sq > 64 ) throw('whatever');
  return 2n ** BigInt(sq - 1);
}

export const total = (): BigInt => {
  let tot = 0n;
  for ( let i = 1; i <= 64; i++ ) {
    tot += square(i);
  }
  return tot;
}
